### ISSUE TITLE
Assert failure: Consistency check failed: AV in clr at this callstack

### ISSUE THREAD

**Issue by VincentBu** [dotnet/coreclr#27604]
**Job:**
[coreclr-corefx #20191031.1 Run Test Pri1 CoreFX Windows_NT x86 checked](https://dev.azure.com/dnceng/public/_build/results?buildId=409206)

**Detail:**
https://helix.dot.net/api/2019-06-17/jobs/5335a869-be77-4373-9e55-7b2ca13a8cb0/workitems/System.Net.NetworkInformation.Functional.Tests/console

**OS & Arch:**
windows x86

**Mode:**
COMPlus_TieredCompilation=0

**Log:**
```
Assert failure(PID 5988 [0x00001764], Thread: 4028 [0x0fbc]): Consistency check failed: AV in clr at this callstack:
------
CORECLR! Object::GetMethodTable + 0x3 (0x72a9c81a)
CORECLR! ArrayBase::GetDataPtr + 0x1F (0x72df228e)
CORECLR! MngdFixedArrayMarshaler::ConvertContentsToManaged + 0x1B2 (0x72f1b882)
<no module>! <no symbol> + 0x0 (0x09d8c520)
CORECLR! CallDescrWorkerInternal + 0x34 (0x72b08091)
CORECLR! CallDescrWorker + 0xD7 (0x72ce3876)
CORECLR! CallDescrWorkerWithHandler + 0x11B (0x72ce39f3)
CORECLR! MethodDescCallSite::CallTargetWorker + 0x5F9 (0x72ce4022)
CORECLR! MarshalStructViaILStub + 0x198 (0x72dd0263)
CORECLR! MarshalNative::PtrToStructureHelper + 0x30C (0x72ed947c)
SYSTEM.PRIVATE.CORELIB! <no symbol> + 0x0 (0x7267558b)
SYSTEM.PRIVATE.CORELIB! <no symbol> + 0x0 (0x72676c73)
<no module>! <no symbol> + 0x0 (0x09d8b931)
<no module>! <no symbol> + 0x0 (0x09d8076b)
<no module>! <no symbol> + 0x0 (0x09d804fb)
<no module>! <no symbol> + 0x0 (0x09e64df9)
CORECLR! CallDescrWorkerInternal + 0x34 (0x72b08091)
CORECLR! CallDescrWorker + 0xD7 (0x72ce3876)
CORECLR! CallDescrWorkerWithHandler + 0x11B (0x72ce39f3)
-----
.AV on tid=0xfbc (4028), cxr=09C8BC64, exr=09C8BC14
FAILED: false

CORECLR! CHECK::Trigger + 0x2FB (0x72a67fde)
CORECLR! CLRVectoredExceptionHandlerPhase3 + 0x25E (0x72b53932)
CORECLR! CLRVectoredExceptionHandlerPhase2 + 0x4E (0x72b535a1)
CORECLR! CLRVectoredExceptionHandler + 0x13C (0x72b53547)
CORECLR! CLRVectoredExceptionHandlerShim + 0x168 (0x72b53b68)
NTDLL! LdrSetAppCompatDllRedirectionCallback + 0x1CB63 (0x779a04c3)
NTDLL! RtlUnwind + 0x1BA (0x7796051a)
NTDLL! KiUserExceptionDispatcher + 0xF (0x77970fff)
CORECLR! MngdFixedArrayMarshaler::ConvertContentsToManaged + 0x1B2 (0x72f1b882)
<no module>! <no symbol> + 0x0 (0x09d8c520)
    File: F:\workspace\_work\1\s\src\vm\excep.cpp Line: 7537
    Image: C:\dotnetbuild\work\5335a869-be77-4373-9e55-7b2ca13a8cb0\Payload\dotnet.exe

exit code -1073740286
```

**Comment by AaronRobinsonMSFT** [comment]
Perhaps fixed with https://github.com/dotnet/coreclr/pull/27562?

**Comment by jkoritzinsky** [comment]
I can't reproduce the failure locally but from the stack trace I believe this will be fixed by #27562
